### PR TITLE
Update `NonNull` pointer provenance methods' documentation

### PR DESCRIPTION
**Pull request by declanvk**
- Add links to equivalent methods on raw pointers

### CODE REVIEW

**Comment by declanvk** [comment]
cc @RalfJung from earlier comment

File: library/core/src/ptr/non_null.rs [hunk 1]
```diff
@@ -289,8 +293,10 @@ impl<T: ?Sized> NonNull<T> {
     ///
     /// This is a convenience for [`with_addr`][Self::with_addr], see that method for details.
     ///
+    /// For more details see the equivalent method on a raw pointer, [`pointer::map_addr`].
```
Comment (RalfJung) [inline library/core/src/ptr/non_null.rs:296]: Hm, now we have two different "for more details"? `pointer::map_addr` says the same thing that it says here, so I think this new sentence here isn't adding anything.

Comment (declanvk) [reply library/core/src/ptr/non_null.rs:296]: I can remove the other "for more details", that one points to the `NonNull::with_addr`.

**Comment by RalfJung** [comment]
Thanks. :)
@bors r+ rollup

**Pull request merged**